storage/buckets/main.go

```diff
@@ -208,15 +212,26 @@ func removeUser(c *storage.Client, bucketName string) error {
 	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
 	defer cancel()
 	bucket := c.Bucket(bucketName)
-	policy, err := bucket.IAM().Policy(ctx)
+	policy, err := bucket.IAM().V3().Policy(ctx)
 	if err != nil {
 		return err
 	}
 	// Other valid prefixes are "serviceAccount:", "user:"
 	// See the documentation for more values.
 	// https://cloud.google.com/storage/docs/access-control/iam
-	policy.Remove("group:cloud-logs@google.com", "roles/storage.objectViewer")
-	if err := bucket.IAM().SetPolicy(ctx, policy); err != nil {
+	for _, binding := range policy.Bindings {
+		if binding.Role == "roles/storage.objectViewer" {
+			// Filter out member.
+			i := 0
+			for _, member := range binding.Members {
+				if member != "group:cloud-logs@google.com" {
+					binding.Members[i] = member
+				}
+				binding.Members = binding.Members[:i]
```

Comment:
Thought maybe we could find the index of the member then slice it out (https://play.golang.org/p/grn8ypJz0cx), but I think this is better because it would handle multiple instances of the member in the slice? A little hard to wrap my mind around assigning to `binding.Members[:i]` on every loop iteration.

Reply:
Maybe lets just create a new filtered slice rather than editing this one in place? This logic is difficult to follow, especially for a sample. Copying might not be as efficient, but it's easier to read.